nanoFirmwareFlasher.Library/OutputWriter.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+
+namespace nanoFramework.Tools.FirmwareFlasher
+{
+    /// <summary>
+    /// Writes output to the console. In unit tests the output can be directed to
+    /// a per-test implementation to make the library and tool testable.
+    /// This is the easiest way, as library and tool were infested with Console.*
+    /// that is hard to test, especially when running tests in parallel.
+    /// </summary>
+    public static class OutputWriter
+    {
+        #region Fields
+        private static readonly AsyncLocal<IOutputWriter> s_outputWriter = new();
+        #endregion
+
+        #region Properties
+        /// <summary>
+        /// Get or set the foreground color
+        /// </summary>
+        public static ConsoleColor ForegroundColor
+        {
+            get => s_outputWriter.Value?.ForegroundColor ?? Console.ForegroundColor;
+            set
+            {
+                if (s_outputWriter.Value is null)
+                {
+                    Console.ForegroundColor = value;
+                }
+                else
+                {
+                    s_outputWriter.Value.ForegroundColor = value;
+                }
+            }
+        }
+        #endregion
+
+        #region Methods
+        /// <summary>
+        /// Write a line to the standard output
```

Comment:
Missing full stop on most summary comments.

Reply:
Yeah, there should be an .editorconfig rule for that :-)

Reply:
Totally agree. The Styelcop rules have this.